driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCTypesTest.java

```diff
@@ -0,0 +1,428 @@
+        CrateTestServer server = testCluster.randomServer();
+        connectionStr = String.format("crate://%s:%s/", server.crateHost(), server.psqlPort());
+        connection = DriverManager.getConnection(connectionStr);
+        setUpTables();
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        tearDownTables();
+    }
+
+    private static void tearDownTables() throws SQLException {
+        ResultSet rs = connection.createStatement().executeQuery("select schema_name, table_name " +
+                "from information_schema.tables where schema_name " +
+                "not in ('pg_catalog', 'sys', 'information_schema', 'blob')");
+        while (rs.next()) {
+            connection.createStatement().execute(String.format("drop table if exists \"%s\".\"%s\"",
+                    rs.getString("schema_name"), rs.getString("table_name")));
+        }
+    }
+
+    public static void setUpTables() throws InterruptedException, SQLException {
+        connection.createStatement().execute("create table if not exists test (" +
+                " id integer primary key," +
+                " string_field string," +
+                " boolean_field boolean," +
+                " byte_field byte," +
+                " short_field short," +
+                " integer_field integer," +
+                " long_field long," +
+                " float_field float," +
+                " double_field double," +
+                " timestamp_field timestamp," +
+                " object_field object as (\"inner\" string)," +
+                " ip_field ip" +
+                ") clustered by (id) into 1 shards with (number_of_replicas=0)");
+        waitForShards();
+
+        connection.createStatement().execute("create table if not exists arrayTest (" +
+                " id integer primary key," +
+                " str_array array(string)," +
+                " bool_array array(boolean)," +
+                " byte_array array(byte)," +
+                " short_array array(short)," +
+                " integer_array array(integer)," +
+                " long_array array(long)," +
+                " float_array array(float)," +
+                " double_array array(double)," +
+                " timestamp_array array(timestamp)," +
+                " obj_array array(object)" +
+                ") clustered by (id) into 1 shards with (number_of_replicas=0)");
+        waitForShards();
+    }
+
+    private static void waitForShards() throws InterruptedException, SQLException {
+        while (countUnassigned() > 0) {
+            Thread.sleep(100);
+        }
+    }
+
+    private static Long countUnassigned() throws SQLException {
+        ResultSet rs = connection.createStatement()
+                .executeQuery("SELECT count(*) FROM sys.shards WHERE state != 'STARTED'");
+        rs.next();
+        return rs.getLong(1);
+    }
+
+    @Before
+    public void before() throws SQLException {
+        insertIntoTable();
```

Comment:
we don't need to insert data into the table before each test, move it to beforeClass